src/ui/widget.js

```diff
@@ -171,21 +206,30 @@ var widget = function (arg) {
         // if the property is a number, add px to it, otherwise set it to the
         // specified value.  Setting a property to null clears it.  Setting to
         // undefined doesn't alter it.
-        if (/^\s*(\-|\+)?(\d+(\.\d*)?|\d*\.\d+)([eE](\-|\+)?\d+)?\s*$/.test(position[cssAttr])) {
+        if (/^\s*(-|\+)?(\d+(\.\d*)?|\d*\.\d+)([eE](-|\+)?\d+)?\s*$/.test(position[cssAttr])) {
```

Comment:
whats change here? sorry it is hard to debug

Reply:
I removed some needless escaping; that is all.